### ISSUE TITLE
Socket: too many open files

### ISSUE THREAD

**Issue by marklocker** [geckoboard/sql-dataset#10]
Hello, I'm using `sql-dataset-linux-amd64` and when I leave it running for a while I get the following errors:

```
Dataset 'dataset_name' errored: Database query failed: dial tcp: lookup db_host on 8.8.4.4:53: dial udp 8.8.4.4:53: socket: too many open files
```

I'm no expert, but a quick Google suggests that sockets might not be getting closed correctly and it's exhausting the OS limit.

**Comment by jnormington** [comment]
Thanks @marklocker for raising the issue - we wasn't using the sql.Open as golang had intended this fixes that now in the latest PR https://github.com/geckoboard/sql-dataset/pull/11 - this should now be fixed in the latest release v0.1.2 which you can now download from the readme.

**Comment by marklocker** [comment]
Thanks @jnormington, I'll download from the README now and give it a go!
